db/migrate/20161014171034_add_directory_items_indexes.rb

```diff
@@ -0,0 +1,11 @@
+class AddDirectoryItemsIndexes < ActiveRecord::Migration
+  def change
```

Comment:
How is this table updated? This indexes will make write operations slower. 

If the read/write ratio is more on the read side it's OK, just making sure.